### DISCUSSION TITLE
[Input InputLeftElement] How to style when input is focused/disable

### DISCUSSION THREAD

**Discussion by shiranZe** [chakra-ui/chakra-ui#5242] [Q&A] [answered]
I need to style the `InputLeftElement` when the input is focused/disabled.
I've managed to do it with InputRightElement, but not with left one.
I think it's because css hierarchy.
It would have been nice if InputLeftElement, InputRightElement would be a part of the input for these kind of cases.

Here is what I did for the right element:

  ```
const elementStyle = {
    '.chakra-input:focus + .chakra-input__right-element svg': {
      color: 'black',
    },
  };

  return (
    <InputGroup sx={elementStyle}>
      <InputLeftElement ps={4} w="fit-content" h="full" color="grey.600">
        <Svg/>
      </InputLeftElement>
      <Input {...rest} ps={12} variant={Consts.VARIANT.filled} />
      <InputRightElement w="fit-content" pe={4} h="full" color="grey.600">
        <Svg />
      </InputRightElement>
    </InputGroup>
  );
}; 
```

for the left one it doesn't work :/

**Comment by primos63** [comment, marked as answer]
If I may ask a question, can you do what you are requesting with simple HTML and CSS?

I don't think so. There is no previous sibling selector in CSS. Putting the left and right icons into the `input` element itself isn't possible which is why there's an `InputGroup` component.

What Chakra does do is it provides the components which allow you to bring your ideas to life.

Here's a [CodeSandbox](https://codesandbox.io/s/chakra-inputgroup-with-color-bu59g?file=/src/components/ColorInputGroup.js) showing a basic example of adding colors to the left and right elements on focus.

**Reply by shiranZe** [reply]
Thanks!

**Comment by veloware** [comment]
I also stumbled across this when trying to style the left and right elements when the input itself has `isDisabled` prop, I've solved it slightly differently, by reordering the nested components

```
const inputGroupStyles = {
    '.chakra-input:disabled ~ .chakra-input__left-element svg': {
      color: '$icon-02',
    },
  };

...

<InputGroup sx={inputGroupStyles}>
      <Input {...args} />
      <InputLeftElement children={<PlusIcon />} />
      <InputRightElement children={<CrossIcon />} />
 </InputGroup>
```

It still positions the left and right elements correctly, as they use absolute positioning

However, there is a caveat; if the left or right element's children is tab-able, the order is not correct.  For me though, I see this as less of a concern as _normally_ any icons that you can interact with (like password view / hide), sits the right, so isn't impacted.

edit - 

This is also configurable at the theme level too, so it can be used across all instances of an input

```
field: {
        '&:disabled ~ .chakra-input__left-element svg': {
          color: '$icon-02',
        },
       // ... more properties for the field
}
```

**Comment by pedrobernardina** [comment]
Check the list of [available pseudos](https://github.com/chakra-ui/chakra-ui/blob/86dd065351d4c5b6a612ae804dbf09cd7fe45799/packages/core/styled-system/src/pseudos.ts). There are some interesting ones, such as `_groupActive`, `_groupDisabled`, `_groupHover` that might help you.

 `__groupHover` works really well for `InputGroup`.
